src/abstractdatatable/reshape.jl

```diff
@@ -227,23 +226,19 @@ unstack(dt::AbstractDataTable, colkey, value) =
 
 function unstack(dt::AbstractDataTable, colkey::Int, value::Int)
     # group on anything not a key or value:
-    g = groupby(dt, setdiff(_names(dt), _names(dt)[[colkey, value]]), sort=true)
+    g = groupby(dt, setdiff(_names(dt), _names(dt)[[colkey, value]]))
     groupidxs = [g.idx[g.starts[i]:g.ends[i]] for i in 1:length(g.starts)]
     rowkey = zeros(Int, size(dt, 1))
     for i in 1:length(groupidxs)
         rowkey[groupidxs[i]] = i
     end
     keycol = NullableCategoricalArray(dt[colkey])
+    levels!(keycol, unique(dt[colkey]))
     valuecol = dt[value]
-    dt1 = dt[g.idx[g.starts], g.cols]
+    dt1 = nullify!(dt[g.idx[g.starts], g.cols])
```

Comment:
No, not necessarily. If we don't nullify it then it will just use be the same column types as provided by the user. With the decision to always return a fully nullable table with outer joins, I thought it would make sense to return a fully nullable table for unstack too, but I can revert this